pkg/kritis/cryptolib/verify_detached.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cryptolib
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/sha512"
+	"crypto/x509"
+	"encoding/asn1"
+	"encoding/pem"
+	"github.com/pkg/errors"
+	"math/big"
+)
+
+func hashPlaintext(plaintext []byte, signingAlg SignatureAlgorithm) []byte {
+	switch signingAlg {
+	case RsaSignPkcs12048Sha256, RsaSignPkcs13072Sha256, RsaSignPkcs14096Sha256, RsaPss2048Sha256, RsaPss3072Sha256, RsaPss4096Sha256, EcdsaP256Sha256:
+		hashed_plaintext := sha256.Sum256(plaintext)
+		return hashed_plaintext[:]
+	case EcdsaP384Sha384:
+		hashed_plaintext := sha512.Sum384(plaintext)
+		return hashed_plaintext[:]
+	case RsaSignPkcs14096Sha512, RsaPss4096Sha512, EcdsaP521Sha512:
+		hashed_plaintext := sha512.Sum512(plaintext)
+		return hashed_plaintext[:]
+	default:
+		return nil
+	}
+}
+
+func verifyDetached(signature []byte, publicKey []byte, signingAlg SignatureAlgorithm, plaintext []byte) error {
```

Comment:
Will this be used only with pkix/jwt signing?  If so, then please mention it so it's clear this has nothing to do with PGP detached signatures.

Reply:
Thats correct. I added a comment.